cpp/include/algorithms.hpp

```diff
@@ -192,6 +192,44 @@ void force_atlas2(GraphCOOView<vertex_t, edge_t, weight_t> &graph,
                   bool verbose                                  = false,
                   internals::GraphBasedDimRedCallback *callback = nullptr);
 
+/**
+ * @brief Finds an approximate solution to the traveling salesman problem (TSP).
+ *        cuGraph computes an approximation of the TSP problem using hill climbing
+ *        optimization.
+ *
+ *        The current implementation does not support a weighted graph.
+ *
+ * @throws                                    cugraph::logic_error when an error occurs.
+ * @param[in] handle                          Library handle (RAFT). If a communicator is set in the
+ * handle, the multi GPU version will be selected.
+ * @param[in] vtx_ptr                         Device array containing the vertex identifiers used
+ * to initialize the route.
+ * @param[out] route                          Device array containing the returned route.
+ * @param[in] x_pos                           Device array containing starting x-axis positions.
```

Comment:
What's the argument ordering logic?

Reply:
I was following a priority ordering on the parameters for the Python API. The cpp layer follows the Python API except for the route that is a returned parameter. I will work on updating the return and use a float and rmm::uvector tuple instead